Add other headers for Authz Bypass (#521)
OWASP/wstg: document/4-Web_Application_Security_Testing/05-Authorization_Testing/02-Testing_for_Bypassing_Authorization_Schema.md

Once the support for the header `X-Original-URL` or `X-Rewrite-URL` was validated then the tentative of bypass against the access control restriction can be leveraged by sending the expected request to the application but specifying a URL "allowed" by the front-end component as the main request URL and specifying the real target URL in the `X-Original-URL` or `X-Rewrite-URL` header depending on the one supported. If both are supported then try one after the other to verify for which header the bypass is effective.

#### 4. Other Headers to Consider

> Often admin panels or administrative related bits of functionality are only accessible to clients on local networks/connections, therefore if may be possible to abuse various proxy or forwarder related HTTP headers to gain access. Some headers and values to test with are:

Reviewer: I don't like the usage of `/`. Just pick the most fitting one, or use `or`.
```suggestion
Often admin panels or administrative related bits of functionality are only accessible to clients on local networks, therefore it may be possible to abuse various proxy or forwarding related HTTP headers to gain access. Some headers and values to test with are:
```

- Headers:
  - `X-Forward-For`
  - `X-Remote-IP`
  - `X-Originating-IP`
  - `X-Remote-Addr`
  - `X-Client-IP`

> - Values

Reviewer: Link local as well? (169.254.0.0/16) This is found a lot in cloud providers and helps with accessing meta-data (remind you of any breaches recently 🙄 )

> - Values

Reviewer: Maybe mention ports to be part of the values

Author: I'll have to research that

Author: ![](https://cloud.githubusercontent.com/assets/1400574/16980451/9644f722-4e98-11e6-9f0e-29b0eeb588f3.png)
